src/modules/navigator/rtl_direct_mission_land.cpp

```diff
@@ -241,6 +241,14 @@ rtl_time_estimate_s RtlDirectMissionLand::calc_rtl_time_estimate()
 	_rtl_time_estimator.update();
 	_rtl_time_estimator.reset();
 
+	if (_vehicle_status_sub.get().is_vtol) {
+		_rtl_time_estimator.setVehicleType(vehicle_status_s::VEHICLE_TYPE_FIXED_WING);
```

Comment:
This is probably accurate in the "normal" case but doesn't it break the case where you RTL as multi-copter with a VTOL? Or does it automatically forward transition anyways now?

Reply:
Yes let's tell the time estimator the correct current vehicle type, not FW if(VTOL).

Reply:
The only remaining one after your change is when it's a VTOL and you land it hardcodes rotary wing:
https://github.com/PX4/PX4-Autopilot/blob/c8ade38ea593b0015e9c8d935e4a4e3b282d25ee/src/modules/navigator/rtl_direct.cpp#L540-L542
I rely on you knowing that this is correct and needed.